From 5654e314c1d9b703618581c290c8c87183b0d2ac Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Mon, 17 Sep 2018 18:20:31 +0800
Subject: [PATCH 4730/5242] MLK-19437: jpeg: fix kernel build fail when set
 JPEG driver be module

commit  910247c2c0dae319f4d7f49b791f457fbaca1f1a from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix kernel build fail when set JPEG driver be module

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/Kconfig        |    1 +
 drivers/media/platform/imx8/Makefile       |    3 ++-
 drivers/media/platform/mxc/capture/Kconfig |    1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/imx8/Kconfig b/drivers/media/platform/imx8/Kconfig
index b8c4a98..bf3a0a8 100644
--- a/drivers/media/platform/imx8/Kconfig
+++ b/drivers/media/platform/imx8/Kconfig
@@ -23,6 +23,7 @@ config IMX8_PARALLEL_CSI
 
 config IMX8_MIPI_CSI2_YAV
 	tristate "IMX8 MIPI CSI2 Controller Yet Another Version"
+	select V4L2_FWNODE
 	default y
 
 config MXC_CAMERA_OV5640_V3
diff --git a/drivers/media/platform/imx8/Makefile b/drivers/media/platform/imx8/Makefile
index 5dc1802f3..097a91b 100644
--- a/drivers/media/platform/imx8/Makefile
+++ b/drivers/media/platform/imx8/Makefile
@@ -10,5 +10,6 @@ obj-$(CONFIG_MXC_CAMERA_OV5640_MIPI_V3) += ov5640_mipi_v3.o
 max9286_gmsl-objs := max9286.o
 obj-$(CONFIG_GMSL_MAX9286) += max9286_gmsl.o
 obj-$(CONFIG_IMX8_MEDIA_DEVICE) += mxc-media-dev.o
-obj-$(CONFIG_IMX8_JPEG) += mxc-jpeg-hw.o mxc-jpeg.o
+mxc-jpeg-enc-dec-objs := mxc-jpeg-hw.o mxc-jpeg.o
+obj-$(CONFIG_IMX8_JPEG) += mxc-jpeg-enc-dec.o
 obj-$(CONFIG_IMX8_HDMI_RX) += hdmi/
diff --git a/drivers/media/platform/mxc/capture/Kconfig b/drivers/media/platform/mxc/capture/Kconfig
index 2aa4e15..040b6a0 100644
--- a/drivers/media/platform/mxc/capture/Kconfig
+++ b/drivers/media/platform/mxc/capture/Kconfig
@@ -6,6 +6,7 @@ config VIDEO_V4L2_MXC_INT_DEVICE
 config VIDEO_MXC_CSI_CAMERA
 	tristate "CSI camera support"
 	depends on VIDEO_MXC_CAPTURE && VIDEO_V4L2
+	select VIDEOBUF2_DMA_CONTIG
 	---help---
 		This is the video4linux2 capture driver based on CSI module.
 
-- 
1.7.9.5

